workspace/workspace.go

```diff
@@ -102,3 +103,53 @@ func (ws Workspace) Locate(exercise string) ([]string, error) {
 	}
 	return paths, nil
 }
+
+func (ws Workspace) SolutionPath(exercise, solutionID string) (string, error) {
+	paths, err := ws.Locate(exercise)
+	if !IsNotExist(err) && err != nil {
+		return "", err
+	}
+
+	return ws.DetermineSolutionPath(paths, exercise, solutionID, PathContainsSolution)
+}
```

Comment:
This is the method that will be used in the download command. It will create a new workspace that is based on the track rather than on the entire Exercism workspace. That way it only finds the directories that contain the exercise in question (e.g. `thingy`, `thingy-2`, `thingy-3`).

The `Locate` method is the one that walks the workspace to find all the directories that match the name plus an optional numeric suffix. We're using it in the open command to find the list of potential solutions to open in the browser.

The last argument to `DetermineSolutionPath` that lets us figure out for a given path whether or not it contains the solution with the given ID. I inject the function so that I can test it without having to create fixtures for all the combinations of paths that I have.